xonsh/tools.py

```diff
@@ -911,6 +911,51 @@ def suggest_commands(cmd, env, aliases):
     return rtn
 
 
+def print_warning(msg):
```

Comment:
There seems to be a lot of overlap in the logic between this and `print_exception()`. Could you maybe add some private functions or similar to reduce the code duplication?

Reply:
Fixed in cdb2514